robottelo/host_helpers/api_factory.py

```diff
@@ -711,6 +713,217 @@ def wait_for_errata_applicability_task(
+
+        Steps:
+            1. get needed entities from arguments, id, name, or instance. Read all as instance.
+            2. publish the content-view if no versions exist, or needs_publish.
+            3. promote the newest content-view-version if not in environment already. Skip for 'Library'.
+            4. assign environment and content-view to the activation-key, if not associated.
+            5. Register the host, using the activation-key associated with content.
+            6. Add a rex_key to the client if desired, enable all available repositories if desired.
+
+        return: dict containing the updated entities:
+            {
+                'client', 'organization', 'activation_key',
+                'environment', 'content_view',
+            }
+        """
+        entities = {
+            'Organization': organization,
+            'ActivationKey': activation_key,
+            'LifecycleEnvironment': environment,
+            'ContentView': content_view,
+        }
+        assert hasattr(client, 'hostname')
+
+        # method for updating entities in dict above,
+        # called after modifying associated entities
+        def read_entities():
+            nonlocal entities
+            entities = {k: v.read() for k, v in entities.items()}
+
+        # for entity arguments matched to above params:
+        # fetch entity instance on satellite
+        #   from given id or name, else read passed argument as an instance.
+        for entity, value in entities.items():
+            param = None
+            # passed int for entity, try to read by id
+            if isinstance(value, int):
+                # equivalent: _satellite_.api.{KEY}(id=VALUE).read()
+                param = getattr(self._satellite.api, entity)(id=value).read()
+            # passed str, search for entity by name
+            elif isinstance(value, str):
+                # search for org name itself, will be just scoped to satellite
+                if entity == 'Organization':
+                    search_query = f'name="{value}"'
+                    # equivalent: _satellite_.api.{KEY}().search(...name={VALUE})
+                    result = getattr(self._satellite.api, entity)().search(
+                        query={'search': search_query}
+                    )
+                # search of non-org entity by name, will be scoped to organization
+                else:
+                    search_query = (
+                        f"name='{value}' and organization_id={entities['Organization'].id}"
+                    )
+                    result = getattr(self._satellite.api, entity)(
+                        organization=entities['Organization']
+                    ).search(query={'search': search_query})
+                if not len(result) > 0:
+                    pytest.fail(
+                        f'Could not find {entity} name: {value}, by search query: "{search_query}"'
+                    )
```

Comment:
Similar to before, this is a helper method, not a fixture or test, come up with an appropriate return that the caller of this method can determine whether a failure is appropriate or not.
This goes for other cases where you're failing a test within this helper method.